apps/src/code-studio/initApp/project.js

```diff
@@ -586,22 +586,40 @@ var projects = (module.exports = {
       this.updateChannels_(callback);
     }
   },
-  setLibraryDetails(newName, newDescription, publishing = undefined) {
+  /**
+   * Updates the current channel's library details.
+   *
+   * @param {Object} config - Object containing library details.
+   * @param {string} config.newName
+   * @param {string} config.newDescription
+   * @param {string} config.newVersionId - S3 version ID for the current library version. Passing this value as -1 will nullify libraryLatestVersion.
+   * @param {boolean} config.publishing - true if library is being published, false if library is being unpublished, undefined otherwise.
+   */
+  setLibraryDetails(config = {}) {
     current = current || {};
+    const {newName, newDescription, newVersionId, publishing} = config;
+
     if (
       current.libraryName !== newName ||
       current.libraryDescription !== newDescription ||
+      current.latestLibraryVersion !== newVersionId ||
       publishing !== undefined
```

Comment:
Now that `setLibraryDetails` is becoming more generic, it may make sense to split this if condition up. i.e. 
```
if (current.libraryName !== newName) {
  current.libraryName = newName
}
if (current.libraryDescription !== newDescription) {
  current.libraryDescription = newDescription
}
...
```

Reply:
yeah, i was having trouble with this. i like that we only call `this.updateChannels_()` if something has actually changed, but i think it's reasonable to always call updateChannels, since that's kind of the purpose of this method